server/lib/src/idm/account.rs

```diff
@@ -683,6 +704,38 @@ impl Account {
             }
         }
     }
+
+    pub fn verify_application_password(
+        &self,
+        app_uuid: &Uuid,
+        app_label: &str,
+        cleartext: &str,
+        ct: Duration,
+    ) -> Result<Option<UserAuthToken>, OperationError> {
+        for ap in self.apps_pwds.values() {
+            if *app_uuid == ap.application && app_label == ap.label {
+                if ap.password.verify(cleartext).map_err(|e| {
+                    error!(crypto_err = ?e);
+                    e.into()
+                })? {
+                    let session_id = uuid::Uuid::new_v4();
+                    let uat: Option<UserAuthToken> = self.to_userauthtoken(
+                        session_id,
+                        SessionScope::ReadWrite,
+                        ct,
+                        DEFAULT_AUTH_SESSION_EXPIRY,
```

Comment:
We might need a separate session expiry here depending on the application in question? @yaleman Is session time on ldap passwords something we should have configurable in future?